### DISCUSSION TITLE
[Feature Request] Improve appearance of Latin alphabet text / fonts

### DISCUSSION THREAD

**Discussion by pan4b** [Floorp-Projects/Floorp#405] [Feature Request]
### Is your feature request related to a problem?

Latin alphabet text looks jagged. Small fonts and bold fonts are especially problematic. (This issue doesn't seem to affect CJK text.)

Screenshot in Floorp 11.2, Spiral Linux XFCE 11.220925. See screenshot at full size to really see the problem.
![Screenshot_2023-08-22_fonts-floorp](https://github.com/Floorp-Projects/Floorp/assets/142930991/37a013df-0a36-491d-88f0-6f6c731fc0c4)

### Describe the solution you'd like

Please give an option to make Latin alphabet fonts look better (I'm not sure how; maybe default option in settings like "Fonts / Turn on anti-aliasing"?).

Here's a screenshot of the same page in Firefox 102.14.0esr, Spiral Linux XFCE 11.220925. Latin fonts look much better. (Again you need to see the image at full size to really appreciate the difference.)
![Screenshot_2023-08-22_fonts-firefox](https://github.com/Floorp-Projects/Floorp/assets/142930991/b58550a1-a6a6-4834-9e1f-516baf513430)

### Additional context

Floorp is such a beautiful browser. It would be nice if the text on webpages was not so ugly by default.

**Comment by Ryosuke-Asano** [comment]
On Linux, It may affect by operating system.

Did you try to change Floorp's font?

**Reply by pan4b** [reply]
Hi Ryosuke, thanks for your response.

I also wondered about that, but I'm comparing Firefox ESR and Floorp on the same system, so the same fonts should be availabe to both, right?

I wondered if the font settings of Firefox and Floorp were different, but (as far as I can tell) they are not. 

Same system, same fonts, same settings... different results. Weird.

I finally remembered about "about:config" and I wondered if maybe something is different there.

about:config / antialias settings are the same in Firefox and Floorp.

- gfx.font_rendering.ahem_antialias_none: false
- webgl.default-antialias: true

So many variables in "about:config" for 'font', not sure if I can check them all. (Is there a way that software could compare and highlight any differences?)

The only thing I've noticed so far is that Floorp has an item called "layout.cc.font-palette.enabled" (set to 'true') and FF (ESR) doesn't list that item at all. But I have no idea if this is related to my problem.

Here are the default Settings for both Firefox and Floorp:

- **Fonts for Latin**
- Default font: DejaVu Serif, size 16
- Proportional: Serif, size 16
- Serif: Default (DejaVu Serif)
- Sans Serif: Default (DejaVu Sans)
- Monospace: Default (DejaVu Sans Mono) size 12
- Minimum font size: None
- Allow pages to choose own fonts: Yes

I changed the fonts to:

- Serif: Georgia
- Sans Serif: Arial
- Monospace: Courier New

That seems to make no difference to the jagged font problem in Floorp, not after refreshing the webpage and not even after closing and restarting Floorp. I still see the same jagged font. (I also made the same changes to the font Settings in FF, after a page refresh the fonts were different.)

Trying one more thing: unticking "Allow pages to choose their own fonts [etc]"... and that works! Immediate change, no more jagged font!

Strange that having it set to "Allow pages..." doesn't cause a problem in FF but it does in Floorp.

I hope unticking "Allow pages..." doesn't cause any other problems with viewing webpages, but at the moment it has stopped the jagged font problem.

Until the origin of the jagged font problem is found and resolved, perhaps Floorp (at least for Linux) should should come with "Allow pages..." unticked by default.

Thanks for making Floorp, it is now my default browser! :)

**Reply by Ryosuke-Asano** [reply]
> I also wondered about that, but I'm comparing Firefox ESR and Floorp on the same system, so the same fonts should be availabe to both, right?

It's true.

> So many variables in "about:config" for 'font', not sure if I can check them all. (Is there a way that software could compare and highlight any differences?

I don't know about that. But, prefs.js has all information about your preferences (at your profile).


> The only thing I've noticed so far is that Floorp has an item called "layout.cc.font-palette.enabled" (set to 'true') and FF (ESR) doesn't list that item at all. But I have no idea if this is related to my problem.

Floorp has never changed the option. It's odd.


Your situation is a rare case. Maybe affected by third-party software or operating system. Did you check add-on?

**Comment by mrguyperson** [comment]
I have the same issue. I'm on OpenSUSE Tumbleweed and installed the flatpak version with no extensions. I noticed that some websites had thin, pixelated looking text, while other pages looked fine. I went into font settings, selected Advanced, and then unchecked "Allow pages to choose their own fonts, instead of your selections above". This seems to have fixed the issue, at least on the sites I checked (Facebook, for example). Maybe this is helpful!
